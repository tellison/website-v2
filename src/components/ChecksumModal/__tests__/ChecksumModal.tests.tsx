import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import ChecksumModal from '..';

Object.assign(navigator, {
    clipboard: {
      writeText: vi.fn(),
    },
});

const navigatorClipboardSpy = vi.spyOn(navigator.clipboard, 'writeText');

describe('ChecksumModal component', () => {
  it('ChecksumModal renders correctly', () => {
    const { container } = render(
      <ChecksumModal />
    );
    expect(container).toMatchSnapshot();
  });

  it('ChecksumModal copies correctly', async () => {
    const { container } = render(
      <ChecksumModal />
    );
    navigatorClipboardSpy.mockImplementationOnce(() => Promise.resolve());
    userEvent.click(screen.getByText('Copy'));
    await screen.findByText('Copied');
    expect(container).toMatchSnapshot();
    expect(navigatorClipboardSpy).toHaveBeenCalledTimes(1);
  });
});
