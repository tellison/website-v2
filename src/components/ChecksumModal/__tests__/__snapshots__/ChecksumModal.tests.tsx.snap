// Vitest Snapshot v1

exports[`ChecksumModal component > ChecksumModal copies correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby="checksumModalLabel"
    class="modal fade"
    id="checksumModal"
    style="z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-lg"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="checksumModalLabel"
          >
            Checksum (SHA256)
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body text-start"
        >
          <p>
            The checksum is used to ensure the file has not been corrupted during download. You should compare the checksum of the file you received with the value below to ensure the file you received is complete and unmodified.
          </p>
          <p>
            On Windows use 
            <code>
              certUtil -hashfile 
              <var>
                file
              </var>
               SHA256
            </code>
            , on Linux use 
            <code>
              sha256sum 
              <var>
                file
              </var>
            </code>
            , and on macOS use 
            <code>
              shasum -a 256 
              <var>
                file
              </var>
            </code>
            .
          </p>
          <input
            readonly=""
            style="width: 100%;"
            value=""
          />
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            Copied
          </button>
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ChecksumModal component > ChecksumModal renders correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby="checksumModalLabel"
    class="modal fade"
    id="checksumModal"
    style="z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-lg"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="checksumModalLabel"
          >
            Checksum (SHA256)
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body text-start"
        >
          <p>
            The checksum is used to ensure the file has not been corrupted during download. You should compare the checksum of the file you received with the value below to ensure the file you received is complete and unmodified.
          </p>
          <p>
            On Windows use 
            <code>
              certUtil -hashfile 
              <var>
                file
              </var>
               SHA256
            </code>
            , on Linux use 
            <code>
              sha256sum 
              <var>
                file
              </var>
            </code>
            , and on macOS use 
            <code>
              shasum -a 256 
              <var>
                file
              </var>
            </code>
            .
          </p>
          <input
            readonly=""
            style="width: 100%;"
            value=""
          />
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            Copy
          </button>
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
